-code-checks

ERROR: Build errors found during clang-tidy run.
```
PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc:148:87: error: no viable overloaded '=' [clang-diagnostic-error]
            if (!useValueMapForPUPPI && aspackedCandidate.isNull()) aspackedCandidate = isocand;
                                                                                      ^
/cvmfs/cms-ib.cern.ch/nweek-02630/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-05-29-2300/src/DataFormats/Common/interface/Ptr.h:40:9: note: candidate function (the implicit copy assignment operator) not viable: no known conversion from 'Ptr<reco::Candidate>' to 'const Ptr<pat::PackedCandidate>' for 1st argument
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc
+++ b/PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc
@@ -139,13 +139,14 @@ namespace citk {
       }
       for (size_t ic = 0; ic < isolate_with->size(); ++ic) {
         auto isocand = isolate_with->ptrAt(ic);
-        edm::Ptr<pat::PackedCandidate> aspackedCandidate; 
+        edm::Ptr<pat::PackedCandidate> aspackedCandidate;
         auto isotype = helper.translatePdgIdToType(isocand->pdgId());
         const auto& isolations = _isolation_types[isotype];
         for (unsigned i = 0; i < isolations.size(); ++i) {
           if (isolations[i]->isInIsolationCone(cand_to_isolate, isocand)) {
             double puppiWeight = 0.;
-            if (!useValueMapForPUPPI && aspackedCandidate.isNull()) aspackedCandidate = edm::Ptr<pat::PackedCandidate>(isocand);
+            if (!useValueMapForPUPPI && aspackedCandidate.isNull()) 
+              aspackedCandidate = edm::Ptr<pat::PackedCandidate>(isocand);
             if (!useValueMapForPUPPI && !usePUPPINoLepton)
               puppiWeight = aspackedCandidate->puppiWeight();  // if miniAOD, take puppiWeight directly from the object
             else if (!useValueMapForPUPPI && usePUPPINoLepton)

--- a/PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc
+++ b/PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc
@@ -145,7 +145,7 @@ namespace citk {
         for (unsigned i = 0; i < isolations.size(); ++i) {
           if (isolations[i]->isInIsolationCone(cand_to_isolate, isocand)) {
             double puppiWeight = 0.;
-            if (!useValueMapForPUPPI && aspackedCandidate.isNull()) aspackedCandidate = isocand;
+            if (!useValueMapForPUPPI && aspackedCandidate.isNull()) aspackedCandidate = edm::Ptr<pat::PackedCandidate>(isocand);
             if (!useValueMapForPUPPI && !usePUPPINoLepton)
               puppiWeight = aspackedCandidate->puppiWeight();  // if miniAOD, take puppiWeight directly from the object
             else if (!useValueMapForPUPPI && usePUPPINoLepton)

--- a/RecoMuon/MuonIsolation/plugins/MuonPFIsolationWithConeVeto.cc
+++ b/RecoMuon/MuonIsolation/plugins/MuonPFIsolationWithConeVeto.cc
@@ -41,9 +41,11 @@ DEFINE_EDM_PLUGIN(CITKIsolationConeDefinitionFactory, MuonPFIsolationWithConeVet
 
 bool MuonPFIsolationWithConeVeto::isInIsolationCone(const reco::CandidatePtr& physob,
                                                     const reco::CandidatePtr& iso_obj) const {
-  if (iso_obj->pt() <= _vetoThreshold ) return false;
+  if (iso_obj->pt() <= _vetoThreshold )
+    return false;
   const double deltar2 = reco::deltaR2(*physob, *iso_obj);
-  if (deltar2 <= _vetoConeSize2 || deltar2 >= _coneSize2) return false;
+  if (deltar2 <= _vetoConeSize2 || deltar2 >= _coneSize2)
+    return false;
   
   //the rest will check the vertex selection
   const pat::PackedCandidatePtr aspacked(iso_obj);

